### PR TITLE
MAT-8567: Add Dedupe to requestallvaluesetsexpansions

### PR DESCRIPTION
**Pull request by ethankaplan**
## Terminology Service PR

Jira Ticket: [MAT-8567](https://jira.cms.gov/browse/MAT-8567)
(Optional) Related Tickets:

### All Submissions
* [ ] This PR has the JIRA linked.
* [ ] Required tests are included.
* [ ] No extemporaneous files are included (i.e Complied files or testing results).
* [ ] This PR is merging into the **correct branch**.
* [ ] All Documentation needed for this PR is Complete (or noted in a TODO or other Ticket).
* [ ] Any breaking changes or failing automations are noted by placing a comment on this PR.

### DevSecOps
If there is a question if this PR has a security or infrastructure impact, please contact the Security or DevOps engineer assigned to this project to discuss it further.

* [ ] This PR has NO significant security impact (i.e Changing auth methods, Adding a new user type, Adding a required but vulnerable package).

### Reviewers
By Approving this PR you are attesting to the following:

*  Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose.
*  The tests appropriately test the new code, including edge cases.
*  If you have any concerns they are brought up either to the developer assigned, security engineer, or leads.

### CODE REVIEW

File: src/main/java/gov/cms/madie/terminology/service/FhirTerminologyService.java [hunk 1]
```diff
@@ -97,27 +95,23 @@ public List<ValueSet> requestAllValueSetsExpansions(
...
-      // exists
-      // This happens if value set has multiple page requests.
       if (existingValueSet == null) {
         allValueSets.add(valueSetResource);
       } else {
-        existingValueSet
-            .getExpansion()
-            .getContains()
-            .addAll(valueSetResource.getExpansion().getContains());
+        List<ValueSet.ValueSetExpansionContainsComponent> newContains =
```
Comment (jkotanchik-SB) [inline src/main/java/gov/cms/madie/terminology/service/FhirTerminologyService.java]: `new` prefixes can be a missed opportunity to provide more context to future devs. Unless the code is building a new object, recommend using a more descriptive prefix. In this case, maybe `dedupedContains`.

**Review by jkotanchik-SB — APPROVED** [review]
